lib/nats.js

```diff
@@ -797,6 +801,23 @@ Client.prototype.processInbound = function() {
 	if (client.checkTLSMismatch() === true) {
 	  return;
 	}
+
+  // Always try to read the connect_urls from info
+  if(client.info.connect_urls && client.info.connect_urls.length > 0) {
+    // don't add duplicates
+    var known = [];
+    client.servers.forEach(function(server) {
+      known.push(server.url.href);
+    });
+    // add new ones
+    client.info.connect_urls.forEach(function(server) {
+      var u = 'nats://' + server;
+      if(known.indexOf(u) === -1) {
+        client.addServer(u);
```

Comment:
Yes it does, it is calling addServer which handles randomization options:
https://sourcegraph.com/github.com/nats-io/node-nats@b5db73ee06428471568a730c3b0ff7f907fa6943/-/blob/lib/nats.js#L990:18